### DISCUSSION TITLE
newbie question - how to link lights to an area in the floorplan

### DISCUSSION THREAD

**Discussion by javixeneize** [ExperienceLovelace/ha-floorplan#202] [Q&A] [answered]
Hi

Apologies for the newbie question, i am doing my first steps with floorplans.

I have created my floorplan. I have added the css and the svg for the light, and also the yaml , changing the entity name to a light entity in my HA. Moreover, i followed the instructions here
 - https://experiencelovelace.github.io/ha-floorplan/docs/example-light/
 
Now my question is... how do i reflect that in my floorplan? It is, what i want is to increase/decrease the brightness of the room when the light is on, as in this example, https://experiencelovelace.github.io/ha-floorplan/docs/example-floorplanner-home/

Thanks!

**Comment by OzGav** [comment, marked as answer]
You will need to provide more info about the problem you are having as if you replicated the example it should work. The yaml has the rules for what css changes are to occur with a state change or a tap_action. The css changes are what makes the look of your floorplan change.

**Reply by javixeneize** [reply]
I think the problem is related to the svg. 

This is my card:

type: custom:floorplan-card
full_height: true
config:
  style: null
  image: /local/floorplan/examples/home/ground.svg
  stylesheet: /local/floorplan/examples/home/home.css
  log_level: info
  console_log_level: info
  defaults:
    hover_action: hover-info
    tap_action: more-info
  rules:
    - entity: light.salon_pie
      element: light.salon_pie
      state_action:
        action: call-service
        service: floorplan.image_set
        service_data:
          image: /local/floorplan/examples/home/light_salon_pie.svg
          cache: true
      tap_action: toggle


I got this svg, https://raw.githubusercontent.com/ExperienceLovelace/ha-floorplan/master/docs/_docs/floorplan/examples/light/light.svg and replaced light.garage with light.salon_pie, as it is the name of my entity

Looking at the log, there is a warning

WARNING CONFIG Cannot find element 'light.salon_pie' in SVG file

If there is any more info needed, let me know

Thanks

**Reply by OzGav** [reply]
Yes you need to open up the SVG find the object labeled light.garage and change that to light.salon_pie. 

P.S. when posting code please use highlight it all and press the <> button to format it and make it easier to read.

**Reply by javixeneize** [reply]
apologies, didnt realise there is a button to do that
![light_salon_pie](https://user-images.githubusercontent.com/12761574/152758177-684a7b22-2876-439c-b1fd-4d999c41a510.svg)

I changed all the occurences in the svg, and still same error. I have attached the file

**Reply by OzGav** [reply]
OK I am a little confused as there is no `light.garage` in the original light example SVG. Also this is a more advanced example however it will work for you assuming your bulb supports the required attributes. Is your light.salon_pie an RGB bulb? If it isn't then look at the [home ](https://experiencelovelace.github.io/ha-floorplan/docs/example-home/) example and the garage light in there.

Also you can run into problems with your browser caching the images. If you use chrome then F12 to open the developer tools, go to the network tab and make sure DISABLE CACHE is checked. Leave that open and reload the page.

**Reply by javixeneize** [reply]
I tried that in an incognito tab and same problem. Sorry, that svg was light.office, should be the same isnt it? 

The bulb is rgb, yes

I have modified the card, leaving jst this

`type: custom:floorplan-card
full_height: true
config:
  style: null
  image: /local/floorplan/examples/home/ground.svg
  stylesheet: /local/floorplan/examples/home/home.css
  log_level: info
  console_log_level: info
  defaults:
    hover_action: hover-info
    tap_action: more-info
  rules:
    - entity: light.salon_pie
      element: light.salon_pie
`

And it still has the error

Maybe is the floorplan svg the one that is triggering this error and not the light one?

**Reply by OzGav** [reply]
OK not sure what you have changed but in your initial post you have your image path as ` /local/floorplan/examples/home/ground.svg` so that should be the file you are editing. This is the file that must have an object which has the ID of `light.salon_pie`

**Reply by javixeneize** [reply]
ah, ok, i get it now. So, the floorplan svg should have the ID of the area, isnt it?

Thanks

**Reply by OzGav** [reply]
The yaml and SVG and CSS all work together. The SVG has a bunch of objects in it. You can then reference those objects in your YAML and change their CSS properties to create the effect you desire. The CSS file just allows you to have a bunch of predefined classes so you can use the `floorplan.class_set` call rather than using `floorplan.style_set` and having to repeat all the style changes for every rule.

**Reply by javixeneize** [reply]
Thanks! Will try it later

**Reply by javixeneize** [reply]
Hi

I tried that and the error disappeared. I have modified my svg renaming my room as light.salon_pie (home_modified.svg). By clicking on the area, i can switch on and off the light, so this is progress. However, it doesnt light the room in the floorplan as it is doing in the demo

This is the new yaml i have now


```
type: custom:floorplan-card
full_height: true
config:
  style: null
  image: /local/floorplan/examples/home/home_modified.svg
  stylesheet: /local/floorplan/examples/home/home.css
  log_level: info
  console_log_level: info
  defaults:
    hover_action: hover-info
    tap_action: more-info
  rules:
    - entity: light.salon_pie
      element: light.salon_pie
      tap_action: light.toggle
      state_action:
        service: floorplan.class_set
        service_data: '${(entity.state === "on") ? "light-on" : "light-off"}'

```


I am assuming i just need an svg image for my floorplan and the css will make the area brighter or darker, right?

Thanks

**Reply by OzGav** [reply]
Yes so you need to have classes in `home.css` called `light-on` and `light-off` that apply a style to show or hide an object. 

In your SVG you should (typically) have a background image which is all lights off. On top of this you will have objects (could be a PNG image, could be a SVG rectangle) that are given the ID of your entities. So it is the light on image object which you actually are touching. When that object is hidden (i.e. the entity is off) you see the background image. When the entity is switched on then the SVG object becomes visible and obscures the part of the light off image below and thus makes it look like the light has come on.

**Reply by OzGav** [reply]
For more info look at these discussions

https://github.com/ExperienceLovelace/ha-floorplan/discussions/116
https://community.home-assistant.io/t/floorplan-now-available-as-a-lovelace-card/115489/416?u=ozgav
